dask_expr/io/parquet.py

```diff
@@ -759,8 +929,23 @@ def _filtered_task(self, index: int):
             self.columns,
             self.filters,
             self._dataset_info["schema"],
+            self.index.name if self.index is not None else None,
         )
 
+    @property
+    def _fusion_compression_factor(self):
+        if self.operand("columns") is None:
+            return 1
+        approx_stats = self.approx_statistics()
+        total_uncompressed = 0
+        for col in approx_stats["columns"]:
+            total_uncompressed += col["total_uncompressed_size"]
+        after_projection = 0
+        for col in self.operands("columns"):
+            after_projection += col["total_uncompressed_size"]
+
+        return max(after_projection / total_uncompressed, 0.001)
```

Comment:
@phofl this now calculates the approx compression factor of the columns using parquet statistics.
The parquet statistics here originate from a sample of files. Currenlty I'm sampling the smallest, largest and median (based on file size) and am averaging them. This aggregation/approximation strategy can be adjusted of course but I figured I'll start somewhere and this isn't horribly, I hope.

With this sample on a scale 1k TPCH dataset this gives these statistics

![image](https://github.com/dask-contrib/dask-expr/assets/8629629/55a85829-89eb-4dea-8cad-e1b1f27d2522)

Therefore, whenever comments are dropped, this will cause quite drastic compression. I'll shortly run some A/B tests

Reply:
so, first prelim results show that the compaction is indeed much stronger which causes me to run into memory failures.

Reply:
The estimation is not too bad though.

above partition size calculated using memory_usage(deep=True), below the approximated size given the fusion compression factor

![image](https://github.com/dask-contrib/dask-expr/assets/8629629/177faea4-2715-4503-94d6-3f6f0a524f4a)

Reply:
the out of memory issues I run into seem to be a downstream memory blow up and not connected to the IO itself https://github.com/coiled/benchmarks/issues/1430